src/main/java/com/ericsson/ei/handlers/StatusHandler.java

```diff
@@ -0,0 +1,222 @@
+        } else {
+            status = Status.UNAVAILABLE;
+
+            LOGGER.error("Available connections in MongoDB is less than expected.\n"
+                    + "Available connections: '{}' used connections: '{}' expected available connections: '{}'",
+                    availableConnections, current, REQUIRED_AVAILABLE_CONNECTIONS);
+        }
+
+        return status;
+    }
+
+    /**
+     * Returns the status of the existing rabbitMQ connection.
+     *
+     * @return
+     */
+    private Status getRabbitMQStatus() {
+        Status status;
+
+        boolean isActive = rmqHandler.getContainer().isActive();
+        boolean isRunning = rmqHandler.getContainer().isRunning();
+        LOGGER.debug("RabbitMQ is running '{}' is active '{}'.", isRunning, isActive);
+
+        if (isActive && isRunning) {
+            status = Status.AVAILABLE;
+        } else {
+            status = Status.UNAVAILABLE;
+        }
+
+        LOGGER.debug("RabbitMQ status is {}", status);
+        return status;
+    }
+
+    /**
+     * Returns the status of the EventRepository service.
+     *
+     * @return
+     */
+    private Status getEventRepositoryStatus() {
+        Status status;
+        if (StringUtils.isEmpty(erUrl)) {
+            status = Status.DISABLED;
+        } else {
+            // /HealtCheck or /Status endpoint must be available in the service to check.
+            // TODO: HTTPRequest should be made to the ER.
+            status = Status.UNKNOWN;
+        }
+
+        LOGGER.debug("EventRepository status is {}", status);
+        return status;
+    }
+
+    /**
+     * Returns the calculated Eiffel Intelligence status.
+     *
+     * @return status
+     */
+    private Status getEiffelIntelligenceStatus() {
+        Status status;
+        boolean availability = getServiceAvailability();
+
+        if (availability) {
+            status = Status.AVAILABLE;
+        } else {
+            status = Status.UNAVAILABLE;
+        }
+
+        LOGGER.debug("Eiffel Intelligence status is {}", status);
+        return status;
+    }
+
+    /**
+     * Check availability on dependent services.
+     *
+     * @return
+     */
+    private boolean getServiceAvailability() {
+        boolean availability;
+
+        boolean mongoDBAvailable = statusData.getMongoDBStatus() == Status.AVAILABLE;
+        boolean rabbitMQAvailable = statusData.getRabbitMQStatus() == Status.AVAILABLE;
+        // TODO: Add check for ER status, may currently only be DISABLED or UNKNOWN.
+        // Once Event repository has a status endpoint the ER status check here should be updated
+        // and this variable added to the availability check.
+        // boolean eventRepositoryAvailable = statusData.getEventRepositoryStatus() !=
+        // Status.DISABLED
+        // && statusData.getEventRepositoryStatus() == Status.AVAILABLE;
+
+        availability = mongoDBAvailable && rabbitMQAvailable;
+        return availability;
```

Comment:
In the meeting we discussed and I was shown a Json object with status info about each components status that is return to the web-UI which showed the status of each component and a fail message per component. What happend with that?

Reply:
It will be shown as a JSON object without messages, after you left the meeting we decided that a message was not needed since the availability status should be enough information.

Reply:
ok